InferStructuralSharing missing in published types (esm/cjs)

### Which project does this relate to?

Router

### Describe the bug

After updating from @tanstack/router 1.129.8 to 1.131.36 I got typescript compilation errors  for the distributed types:

```
node_modules/@tanstack/react-router/dist/esm/index.d.ts:49:36 - error TS2305: Module '"./typePrimitives.js"' has no exported member 'InferStructuralSharing'.

49 export type { ValidateLinkOptions, InferStructuralSharing, ValidateUseSearchOptions, ValidateUseParamsOptions, ValidateLinkOptionsArray, } from './typePrimitives.js';
                                      ~~~~~~~~~~~~~~~~~~~~~~
```

When reviewing the node_modules/@tanstack/react-router/dist/esm/typePrimitives.d.ts file, it lacks the InferStructuralSharing type.

Version 1.129.8 includes this type, defined in typePrimitives.d.ts as 

```
/**
 * @internal
 */
export type InferStructuralSharing<TOptions> = TOptions extends {
    structuralSharing: infer TStructuralSharing;
} ? TStructuralSharing : unknown;

```


### Your Example Website or App

https://stackblitz.com/edit/github-h1spuhyh?file=tsconfig.json

### Steps to Reproduce the Bug or Issue

set skipLibCheck: false

```
npm i
node_modules/typescript/bin/tsc --noEmit
```

### Expected behavior

No  typescript errors

### Screenshots or Videos

<img width="789" height="489" alt="Image" src="https://github.com/user-attachments/assets/ee73f3c8-6c9c-471f-9d94-baf50af99ae4" />

### Platform

- Router Version: 1.131.36
- Typescript: any recent version

### Additional context

_No response_

--- a/packages/react-router/src/typePrimitives.ts
+++ b/packages/react-router/src/typePrimitives.ts
@@ -32,7 +32,7 @@ export type ValidateLinkOptions<
 >
 
 /**
- * @internal
+ * @private
  */
 export type InferStructuralSharing<TOptions> = TOptions extends {
   structuralSharing: infer TStructuralSharing

--- a/packages/router-core/src/scroll-restoration.ts
+++ b/packages/router-core/src/scroll-restoration.ts
@@ -346,7 +346,7 @@ export function setupScrollRestoration(router: AnyRouter, force?: boolean) {
 }
 
 /**
- * @internal
+ * @private
  * Handles hash-based scrolling after navigation completes.
  * To be used in framework-specific <Transitioner> components during the onResolved event.
  *

